src/Atc.Rest.Client/Builder/MessageRequestBuilder.cs

```diff
@@ -70,20 +77,43 @@ public IMessageRequestBuilder WithPathParameter<T>(string name, T value)
             where T : struct
             => WithPathParameter(name, value.ToString());
 
-        public IMessageRequestBuilder WithQueryParameter(string name, string? value)
+        public IMessageRequestBuilder WithHeaderParameter(string name, string value)
```

Comment:
We should preserve behavior across `WithXXX` methods, so that when passing a `null` as value parameter means that no headers is set.

See `WithQueryParameter(string name, object? value)`.